Mathlib/Algebra/CharP/Basic.lean

```diff
@@ -91,7 +91,8 @@ theorem exists_add_pow_prime_eq (hp : p.Prime) (x y : R) :
 
 end CommSemiring
 
-variable (R)
+section AddMonoidWithOne
+variable (R) [AddMonoidWithOne R] (p : ℕ)
 
 /-- The generator of the kernel of the unique homomorphism ℕ → R for a semiring R.
 
```

Comment:
The changes to CharP/Basic.lean seem more general and basic than stuff about 3APs, do they not belong in a separate PR?

Reply:
See the PR description, in particular the mention of #12542!